inst/pages/exercises.qmd

```diff
@@ -1763,36 +1771,35 @@ plotReducedDim(tse, dimred = "MDS") +
 #| code-summary: "Show solution"
 
 # Load necessary libraries
-library(SummarizedExperiment)
+library(mia)
 library(sechm)
+library(bluster)
 
-# Generate sample abundance data
-set.seed(123)
-sample_data <- matrix(rnorm(100), nrow = 10)
-rownames(sample_data) <- paste0("Gene", 1:10)
-colnames(sample_data) <- paste0("Sample", 1:10)
-
-# Perform hierarchical clustering
-hclust_result <- hclust(dist(t(sample_data)), method = "complete")
+# Load abundance data
+data("GlobalPatterns", package = "mia")
 
-# Assign cluster information to metadata
-clusters <- cutree(hclust_result, k = 3)
-metadata <- data.frame(Cluster = as.factor(clusters))
-rownames(metadata) <- colnames(sample_data)
+# Here we extract the first 20 taxa and the first 20 samples as an example
+sample_data <- assay(GlobalPatterns)[1:20, 1:20]
 
 # Create a SummarizedExperiment object
-se <- SummarizedExperiment(assays = list(counts = sample_data),
-                           colData = metadata)
+se <- SummarizedExperiment(assays = list(counts = sample_data))
```

Comment:
GlobalPatterns is a already a TreeSE object (and also SE object because all TreeSE are SE)

Unless I am mistaken, you are here first extracting "sample_data" (assay) from the GlobalPatterns TreeSE object, and then reconstructing this again as an SE. Is this necessary? Why not use the original GlobalPatterns directly - you can rename it if necessary, with `tse <- GlobalPatterns` as in most examples.